src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
+  static alive = [];
   // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    Animal.alive = Animal.alive.filter((animal) => animal !== this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      this.die();
+    }
```

Comment:
The `checkHealth` method is correctly checking the health of the animal and calling `die` if the health is less than or equal to 0. This is a good approach as it encapsulates the logic within the `Animal` class.